rest/tests/test_rest.py

```diff
@@ -544,7 +601,131 @@ def test_immutable_fields(self):
         self.assertNotEqual(term_dict['slug'], result['slug'])
         self.assertNotIn('vocabulary', result)
         term = Term.objects.get(slug=result['slug'])
-        self.assertEqual(vocabulary.id, term.vocabulary.id)
+        self.assertEqual(vocab_slug, term.vocabulary.slug)
+
+    def test_immutable_fields_learning_resource(self):
+        """Test immutable fields for term"""
+        tarball_file = self.get_course_single_tarball()
+        import_file(
+            tarball_file, self.repo.id, self.user.id)
+        resource = get_resources(self.repo.id).first()
+        resource.static_assets.add(StaticAsset.objects.first())
+        lr_id = self.get_learning_resources(self.repo.slug)['results'][0]['id']
+
+        lr_dict = {
+            "id": 99,
+            "learning_resource_type": 4,
+            "static_assets": [3],
+            "title": "Getting Help",
+            "description": "description",
+            "content_xml": "...",
+            "materialized_path":
+                "/course/chapter[4]/sequential[1]/vertical[3]",
+            "url_path": "url_path",
+            "parent": 22,
+            "copyright": "copyright",
+            "xa_nr_views": 1,
+            "xa_nr_attempts": 2,
+            "xa_avg_grade": 3.0,
+            "xa_histogram_grade": 4.0
+        }
+
+        # these keys should be different since they are immutable or set by
+        # the serializer
+        result = self.patch_learning_resource(self.repo.slug, lr_id, lr_dict,
+                                              skip_assert=True)
+        self.assertNotEqual(lr_dict['id'], result['id'])
```

Comment:
All these `assertNotEqual` lines could be done in a loop, since the dict keys are all the same. Although as mentioned above, all immutable field tests should be unnecessary.